core/src/main/java/com/twosigma/beaker/core/rest/PluginServiceLocatorRest.java

```diff
@@ -599,11 +600,19 @@ private String hashIPythonPassword(String password, String pluginId, String comm
     // but it would be nice if there were a way to make this explicit. XXX
     bw.write("print(passwd('" + password + "'))\n");
     bw.close();
-    String hash = br.readLine();
-    if (null == hash) {
-      throw new RuntimeException("unable to get IPython hash");
+
+    Object[] array = br.lines().filter(new Predicate<String>() {
```

Comment:
Can this have a bit better name? Calling it "array" is not going to be very helpful to anyone that's going to work with this code later. Maybe calling it something like "hashes" or "linesWithSHA" would be better?